library/common-library/templates/_region.tpl

```diff
@@ -0,0 +1,74 @@
+{{/*
+Return the region that is being used by the user
+*/}}
+{{- define "newrelic.common.region" -}}
+{{- if and (include "newrelic.common.license._usesCustomSecret" .) (not (include "newrelic.common.region._fromValues" .)) -}}
```

Comment:
I may be mistaken, but can we not evaluate the value of the secret to check if it's an US/EU key? We did this in the OTEL chart: https://github.com/newrelic/helm-charts/blob/master/charts/nr-k8s-otel-collector/templates/_endpoint.tpl

Reply:
Let's read that line:
The implementation of `newrelic.common.license._usesCustomSecret` is [here](https://github.com/newrelic/helm-charts/pull/1463/files/15ad3fd74067f54c30ba6858ee580b1aa9989c1e#diff-0c8f64dcb17357dbcfe03584129bc756eacfa7dd46e7c61767e46b6098a65544R64-R68): Returns true if the uses is suing a custom secret for the license instead of writing it explicitly in the `values.yaml`.

Ht implementation of `newrelic.common.region._fromValues` is [here](https://github.com/newrelic/helm-charts/pull/1463/files/15ad3fd74067f54c30ba6858ee580b1aa9989c1e#diff-35ca0af1666eaed8b3d266a1be796dab098eadcbd1582bd095fe4007542e963cR51-R59): Return the region from `values.yaml` if it is specified by the user.

The line were you left the comment basically test that use case you are pointing: If the user is using a secret for the license and the user does not set a region, as we will not be able to compute the region from the licese, [we fail](https://github.com/newrelic/helm-charts/pull/1463/files/15ad3fd74067f54c30ba6858ee580b1aa9989c1e#diff-35ca0af1666eaed8b3d266a1be796dab098eadcbd1582bd095fe4007542e963cR6).

The point onf this is to use the same UX for both charts as we are going to use on the super-agent chart. Take a look at the draft that is waiting for this to be merged: https://github.com/newrelic/helm-charts/pull/1466/files#diff-f752c985c9b04f49290d4a210d58d0981e8197c5825b7747defdc701fa4e1652R36-R54